packages/core/src/lib/filter/v1/index.ts

```diff
@@ -114,16 +116,70 @@ class Filter extends BaseProtocol implements IFilter {
     const subscription: Subscription<T> = {
       callback,
       decoders: decodersArray,
-      pubSubTopic,
+      pubSubTopic: this.pubSubTopic,
     };
     this.subscriptions.set(requestId, subscription);
 
-    return async () => {
-      await this.unsubscribe(pubSubTopic, contentFilters, requestId, peer);
-      this.subscriptions.delete(requestId);
+    return {
```

Comment:
To me it seems strange to return `unsubscribe` and `unsubscribeAll` from the `subscribe` function.

Reply:
You cannot `unsubscribeAll` in a vacuum, you need at least to know the peer id of the remote peer to whom you want to `unsubscribe all`.

There needs to be a way to manage your subscription for a given peer. You might want to subscribe to different peers for different reasons (either for robustness or because you are interested in messages on different shards).

Either way, `subscribe`, `unsubscribe`, `ping` and `unsubscribe` all needs to done with peer id in context.

We can either ask the consumer to pass the peer id every time, or instead, return an object from `subscribe` that allows you to manage your subscription.

With Filter v1, one could only `subscribe` or `unsubscribe`. So `unsubscribe` was returned. But now, one can:
- `unsubscribe` for specific topics
- `subscribe` to new topics
- `ping` for a subscription
- `unsubcribe_all` to all topics

There's a good point here: one should also be able to use `subscribe` to add content topics too.

Reply:
I don't think it's great that we manually populate the object being returned.

Maybe instead, we should have a `Subscription` class that implements all the methods and gets constructed with a `peer id`

Reply:
Note: I realize the original comment is on v1. My comment applies to v2.